redash/tasks/general.py

```diff
@@ -72,16 +72,23 @@ def purge_failed_jobs():
         for queue in Queue.all():
             failed_job_ids = FailedJobRegistry(queue=queue).get_job_ids()
             failed_jobs = Job.fetch_many(failed_job_ids, rq_redis_connection)
-            stale_jobs = [
-                job
-                for job in failed_jobs
-                if job
-                and (datetime.utcnow() - job.ended_at).seconds
-                > settings.JOB_DEFAULT_FAILURE_TTL
-            ]
-
-            for job in stale_jobs:
-                job.delete()
+            stale_jobs = []
+            for failed_job in failed_jobs:
```

Comment:
If you share my thoughts on the other couple of comments, this whole block might be better represented by a filter on `failed_jobs`

Reply:
I'm not sure I follow, what do you mean with "a filter on `failed_jobs`"?

Reply:
I just mean that it feels like `stale_jobs` is just a sub-list of `failed_jobs` that satisfies a predicate. Something like:

```python
is_stale = lambda job: job.ended_at is None or
                       (datetime.utcnow() - job.ended_at).seconds > settings.JOB_DEFAULT_FAILURE_TTL
stale_jobs = filter(is_stale, compact(failed_jobs))
```

Reply:
While you may be right that this is another way to write it, I don't see this as more readable. But it's up to you, feel free to change the patch the way you like it better.

Reply:
I'm fine with keeping @jezdez's implementation as is, as it leaves room for explaining the different steps.

Reply:
👍 

Don't feel strongly either way, but generally I'm more in the camp of having descriptive variable / function / lambda names instead of comments (i.e. `is_stale = worker_died or too_old`). They just expire slower than comments.